controllers/controllers/services/bindings/controller.go

```diff
@@ -126,10 +123,7 @@ func (r *Reconciler) ReconcileResource(ctx context.Context, cfServiceBinding *ko
 		return ctrl.Result{}, err
 	}
 
-	if cfServiceBinding.Annotations == nil {
-		cfServiceBinding.Annotations = map[string]string{}
-	}
-	cfServiceBinding.Annotations[korifiv1alpha1.ServiceInstanceTypeAnnotationKey] = string(cfServiceInstance.Spec.Type)
+	setInstanceTypeAnnotation(cfServiceBinding, string(cfServiceInstance.Spec.Type))
```

Comment:
I would just inline the function. The function is quite an obvious one-liner, having a dedicated function here from my POV makes the code less readable.